test/controllers/VatDetailsControllerSpec.scala

```diff
@@ -64,11 +65,13 @@ class VatDetailsControllerSpec extends ControllerBaseSpec {
     val mockAccountDetailsService: AccountDetailsService = mock[AccountDetailsService]
     val mockDateService: DateService = mock[DateService]
     val mockAuditService: AuditingService = mock[AuditingService]
-    val mockHybridUserPredicate: HybridUserPredicate = new HybridUserPredicate(mockAccountDetailsService, mockServiceErrorHandler)
+    val mockHybridUserPredicate: HybridUserPredicate = new HybridUserPredicate(mockAccountDetailsService, mockServiceErrorHandler, mcc, ec)
```

Comment:
There's a bunch of stuff being created in this trait that already exists in `ControllerBaseSpec`. Can you re-use those vals where possible?